src/hpr_sup.erl

```diff
@@ -46,6 +46,22 @@ start_link() ->
     supervisor:start_link({local, ?SERVER}, ?MODULE, []).
 
 init([]) ->
+    KeyFileName = application:get_env(hpr, key, "/var/data/hpr.key"),
+    ok = filelib:ensure_dir(KeyFileName),
+    Key =
```

Comment:
Can we put the saving and getting of the pub_key closer together?

Reply:
What do you mean?